pcap/pcapio/reader.go

```diff
@@ -27,6 +24,23 @@ type Info struct {
 // For TypePacket, the capture timestamp and the link-layer type of the packet
 // is indicated in the Info return value.
 type Reader interface {
-	Read() ([]byte, Info, error)
+	Read() ([]byte, BlockType, error)
+	Packet([]byte) ([]byte, nano.Ts, layers.LinkType)
 	Offset() uint64
 }
+
+// NewReader returns a Reader by trying both pcap and pcap-ng formats.
+func NewReader(r io.Reader) (Reader, error) {
+	recorder := detector.NewRecorder(r)
+	track := detector.NewTrack(recorder)
+	_, err := NewPcapReader(track)
+	if err == nil {
+		return NewPcapReader(recorder)
+	}
```

Comment:
Can fixed in a follow up pr but if this returns an error because it's an unsupported pcap version, that error should get returned.

Reply:
NewPcapReader() returns the error as the second arg

Reply:
If NewPcapReader returns a "this version of legacy pcap is invalid err" from a legacy pcap, the error will be over-written by whatever error NewNgReader finds.

Reply:
No, if they both fail, then it's ErrCorruptPcap

Reply:
I think this is ok.
GSD